src/test/java/codeu/controller/NotificationServletTest.java

```diff
@@ -0,0 +1,108 @@
+import javax.servlet.http.HttpSession;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class NotificationServletTest{
+  private NotificationServlet NotificationServlet;
+  private HttpSession mockSession;
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+  private EventStore mockEventStore;
+  private UserStore mockUserStore;
+  private User mockUser;
+
+
+  @Before
+  public void setup() {
+    NotificationServlet = new NotificationServlet();
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/notifications.jsp"))
+        .thenReturn(mockRequestDispatcher);
+
+    mockEventStore = Mockito.mock(EventStore.class);
+    NotificationServlet.setEventStore(mockEventStore);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    NotificationServlet.setUserStore(mockUserStore);
+
+    mockUser = Mockito.mock(User.class);
+  }
+
+
+  @Test
+  public void tesDoGet() throws IOException, ServletException {
+    List<Event> fakeEventList = new ArrayList<>();
+
+    UUID fakeUserId = UUID.randomUUID();
+    User fakeUser = new User(fakeUserId, "testUser", "test_password_hash", Instant.ofEpochMilli(1000));
+    List<String> userInformation = new ArrayList<>();
+    userInformation.add(fakeUser.getName());
+    Event userEvent = new Event(UUID.randomUUID(), "User", fakeUser.getCreationTime(), userInformation);
+    fakeEventList.add(userEvent);
+
+    UUID fakeConversationId = UUID.randomUUID();
+    Conversation fakeConversation = new Conversation(fakeConversationId, fakeUserId, "testConversation", Instant.ofEpochMilli(1000));
+    List<String> conversationInformation = new ArrayList<>();
+    conversationInformation.add(fakeUser.getName());
+    conversationInformation.add(fakeConversation.getTitle());
+    Event conversationEvent = new Event(UUID.randomUUID(), "Conversation", fakeConversation.getCreationTime(), conversationInformation);
+    fakeEventList.add(conversationEvent);
+
+    Message fakeMessage = new Message(UUID.randomUUID(), fakeConversationId, fakeUserId, "testMessage", Instant.ofEpochMilli(1000));
+    List<String> messageInformation = new ArrayList<>();
+    messageInformation.add(fakeUser.getName());
+    messageInformation.add(fakeConversation.getTitle());
+    messageInformation.add(fakeMessage.getContent());
+    Event messageEvent = new Event(UUID.randomUUID(), "Message", fakeMessage.getCreationTime(), messageInformation);
+    fakeEventList.add(messageEvent);
+
+    Instant fakeLastSeenTime = Instant.now();
+    mockUser.setLastSeenNotifications(fakeLastSeenTime);
```

Comment:
How about we make the fake events have different times (e.g. 1000, 2000, and 3000), then set the user's last seen time to 1500 or something, then we should expect that the last two events only are returned? This way we're actually testing that the user only sees the new events.